networking_aci/plugins/ml2/drivers/mech_aci/allocations_manager.py

```diff
@@ -348,70 +349,70 @@ def _segmentation_ids(host_config):
+                if hg_config['direct_mode']:
                     continue
-                self._process_host_or_hostgroup(session, hostgroup, hostgroup_config, allocations, level)
-
-        # remove from table unallocated vlans for any unconfigured
-        # physical networks
-        for allocs in allocations.values():
-            for alloc in allocs:
-                if not alloc.network_id:
-                    LOG.debug("Removing segment %(seg_id)s on "
-                              "host/hostgroup"
-                              "%(host)s level %(level)s type %(segment_type)s from pool",
-                              {'seg_id': alloc.segmentation_id,
-                               'host': alloc.host,
-                               'level': alloc.level,
-                               'segment_type': alloc.segment_type})
-                    session.delete(alloc)
-        LOG.info("ACI allocations synced")
-
-    def _process_host_or_hostgroup(self, session, host, host_config, allocations, level):
-        segment_type = host_config['segment_type']
-        segmentation_ids = self._segmentation_ids(host_config)
-        alloc_key = self._allocation_key(host, level, segment_type)
-        if alloc_key in allocations:
-            for alloc in allocations[alloc_key]:
-                try:
-                    # see if segment is allocatable
-                    LOG.info("Check if allocatable")
-                    LOG.info(alloc.segmentation_id)
-                    segmentation_ids.remove(alloc.segmentation_id)
-                except KeyError:
-                    # it's not allocatable, so check if its allocated
-                    if not alloc.network_id:
-                        # it's not, so remove it from table
-                        LOG.debug("Removing segment %(seg_id)s on "
-                                  "host/hostgroup "
-                                  "%(host)s level %(level)s type %(segment_type)s from pool",
-                                  {'seg_id': alloc.segmentation_id,
-                                   'host': host,
-                                   'level': level,
-                                   'segment_type': segment_type})
-                        session.delete(alloc)
-            del allocations[alloc_key]
-
-        for segmentation_id in sorted(segmentation_ids):
-            alloc = AllocationsModel(host=host, level=level, segment_type=segment_type, segmentation_id=segmentation_id,
-                                     network_id=None)
-            session.add(alloc)
+
+                hg_vlans = self._segmentation_ids(hg_config)
+                if hg_name in allocations:
+                    # delete extra allocs
+                    extra_ids = no_net_allocations[hg_name] - hg_vlans
+                    if extra_ids:
+                        LOG.debug("Deleting %s unused allocations for hostgroup %s", len(extra_ids), hg_name)
```

Comment:
These are unused and according to the config also not in the allowed physnet range? If so, let the log line reflect it.